include/boost/hana/fwd/lazy.hpp

```diff
@@ -68,7 +68,11 @@ BOOST_HANA_NAMESPACE_BEGIN
     //! [1]: http://ldionne.com/2015/03/16/laziness-as-a-comonad
 #ifdef BOOST_HANA_DOXYGEN_INVOKED
     template <typename ...>
-    struct lazy;
+    struct lazy {
+        //! Equivalent to `hana::chain`.
+        template <typename ...T, typename F>
+        constexpr auto operator|(lazy<T...>, F);
```

Comment:
I just broke all of these free functions out into a separate file and then realized that maybe you meant that I simply add the `friend` keyword there. Which is preferred?